### DISCUSSION TITLE
"In Progress" vs "Prototype"

### DISCUSSION THREAD

**Discussion by MgSam** [dotnet/csharplang#1971] [Q&A] [answered]
The [Language Feature Status](https://github.com/dotnet/roslyn/blob/master/docs/Language%20Feature%20Status.md) page shows a number of features "In Progress" and a number of features in "Prototype". The [Developing Language Features](https://github.com/dotnet/roslyn/blob/master/docs/contributing/Developing%20a%20Language%20Feature.md) page makes no distinction between in-progress vs prototype. (It also doesn't reflect the meaning of "Implemented" which is only used once.)

What is the difference, if any? Can the Developing Language Features page be updated to accurately reflect the tags you guys are using?

Thanks.

**Comment by gordanr** [comment]
It would be also fine if we somewhere could find explanation of the terms LDM Champ and Reviewer. I couldn’t find any explanation. What is the difference between these roles?

**Comment by gafter** [comment]
The LDM Champion is a member of the language design team who is responsible for keeping the process moving for the proposed change.

The Reviewer is a member of the compiler development team who is responsible for reviewing the compiler changes made to implement the language feature.

**Comment by MgSam** [comment]
@gafter Any feedback on my question?

**Comment by theunrepentantgeek** [comment, marked as answer]
```
#include <std.disclaimer>
```

Based on the discussions I've read here in this repo, I've been assuming the following definitions:

**In Progress** - someone has made a start on writing actual code that's intended to become the final implementation once complete.

**Prototype** - an experiment intended to allow the author (and others) to learn about what's required for a full implementation. 

Prototypes are useful for discovering edge cases and interactions with other features. They're usually thrown away once the learning is complete.

Now, I'm just a lurker here, so these definitions have _precisely_ **zero** official weight.
